packages/web-components/src/components/rux-modal/rux-modal.tsx

```diff
@@ -73,26 +92,27 @@ export class RuxModal {
     handleClick(ev: MouseEvent) {
         const wrapper = this._getWrapper()
         if (ev.composedPath()[0] === wrapper) {
-            this.ruxModalClosed.emit(false)
+            // this.ruxModalClosed.emit(false)
             this.open = false
         }
     }
 
     @Watch('open')
-    validateName(isOpen: boolean) {
-        if (isOpen) {
+    handleOpen(isOpen: boolean) {
+        //? This is preventing the default focusing of a button if the footer slot exists.
+        //? I thought about runninng it anyway, but it could lead to cases where an unintentional button gets focused.
+        if (isOpen && !this.hasFooter) {
             setTimeout(() => {
                 const button = this._getDefaultButton()
-                button && button.focus()
-            })
+                if (button) {
+                    button.focus()
+                }
+            }, 0)
         }
+        this.open ? this.ruxModalOpened.emit() : this.ruxModalClosed.emit()
```

Comment:
Probably the biggest refactor, moved all open and closed event emissions to the watch open decorator. Before this, the closed event was only emitted on an off click or a button choice.